src/shared/bootspec.c

```diff
@@ -880,19 +880,22 @@ static int insert_boot_entry_addon(
                 char *location,
                 char *cmdline) {
 
+        assert(addons);
+
         if (!GREEDY_REALLOC(addons->items, addons->n_items + 1))
```

Comment:
GREEDY_REALLOC_APPEND() ?

Reply:
Hmm, there's no need to first create a `BootEntryAddon` on stack and copy it? `GREEDY_REALLOC_APPEND` seems better suited for existing objects?

Reply:
If you don't like the suggestion, feel free to ignore it.